_targets.R

```diff
@@ -6,10 +6,40 @@ source("3_visualize/src/plot_timeseries.R")
 options(tidyverse.quiet = TRUE)
 tar_option_set(packages = c("tidyverse", "dataRetrieval")) # Loading tidyverse because we need dplyr, ggplot2, readr, stringr, and purrr
 
+nwis_Cd <- '00010'
+p_width <- 12
+p_height <- 7
+p_units <- "in"
+
 p1_targets_list <- list(
+  tar_target(
+    site_data_01427207,
+    download_nwis_site_data(file.path("1_fetch", "out", "nwis_01427207_data.csv"), parameterCd = nwis_Cd),
```

Comment:
You don't need to make a change here, since passing in only the filename and then extracting the site id from the filename within the function works fine, but if you wanted to specify it here instead, you could, either by typing it out twice, for both a filepath argument and site_id argument, like so:

```r
tar_target(
    site_data_01427207,
    download_nwis_site_data(site_id = '01427207', file.path("1_fetch", "out", "nwis_01427207_data.csv"), parameterCd = nwis_Cd),
    format = "file"
  ),
```

Or you can even define site id within the `tar_target()` call, before you call the `download_nwis_site_data()` command:

```r
tar_target(
    site_data_01427207_csv,
    {
      site_id = '01427207`
      out_file = file.path('1_fetch/out', paste0('nwis_', site_id, '_data.csv'))
      download_nwis_site_data(site_num = site_id,
                              out_file = out_file,
                              parameterCd = parameter_code,
                              startDate = start_date,
                              endDate = end_date)
    },
    format = 'file'
  ),
```

Regardless, you're totally right that this is a lot of hardcoding, and means typing out the site number repeatedly. We'll learn about a more efficient, adaptable approach in pipelines-III

Reply:
Looking forward to learning about this in pipelines-3!